agent_api_rest/src/verification/relying_party/redirect.rs

```diff
@@ -7,15 +7,42 @@ use axum::{
     extract::State,
     http::StatusCode,
     response::{IntoResponse, Response},
-    Form,
 };
 use http_api_problem::ApiError;
 
+use anyhow::{self, Result};
+use serde::de::DeserializeOwned;
+use serde_urlencoded;
+use std::collections::HashMap;
+
+fn from_form_urlencoded_string<T: DeserializeOwned>(encoded: &str) -> anyhow::Result<T> {
+    let string_map: HashMap<String, String> =
+        serde_urlencoded::from_str(encoded).map_err(|e| anyhow::anyhow!("Failed to decodde: {}", e))?;
+
+    // Converts the string map to a JSON map, parsing the stringified JSON values.
+    let json_map: serde_json::Map<String, serde_json::Value> = string_map
+        .into_iter()
+        .map(|(k, v)| {
+            let json_value = serde_json::from_str(&v).unwrap_or_else(|_| serde_json::Value::String(v.to_string()));
+            (k, json_value)
+        })
+        .collect();
+
+    // Converts the JSON map to the wanted type.
+    let value = serde_json::Value::Object(json_map);
+    serde_json::from_value(value).map_err(|e| anyhow::anyhow!("Failed to deserialize into target type: {}", e))
+}
```

Comment:
If I'm not mistaken you've added this function to `openid4vc` now, so we don't need this function declaration here anymore :)